v2/tests/git_initializer_test.go

```diff
@@ -16,20 +17,23 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-var defaultConfigFiles = map[string]string{
-	"brigade.js": `
-		const { events } = require("brigadier");
+const (
+	testJobName = "test-job"
+	testTimeout = time.Duration(120 * time.Second)
+)
 
-		events.on("github.com/brigadecore/brigade/cli", "exec", () => {
-			console.log("Hello, World!");
-		});
-	`}
+type testcase struct {
+	name                       string
+	project                    core.Project
+	configFiles                map[string]string
+	expectedJobLogsContains    string
+	expectedJobPhase           core.JobPhase
+	expectedWorkerLogsContains string
+	expectedWorkerPhase        core.WorkerPhase
+	expectedVCSLogsContains    string
+}
```

Comment:
I'm a little worried that test cases constructed this way might be too rigid for future expansion.

Every case is following the same formula: take this projects, create an event for it, check that _these specific things are true_ (i.e. the worker ended up in this phase, the job ended up in this phase, the git init logs contain this phrase, the worker logs contain this other phrase, etc.)

Things get messy if we want to deviate from that formula. So, for instance, suppose we wanted to add a test case with no git initializer? (I know these are the git initializer tests, but I think we both agreed that this was also the seed for a more extensive suite of integration tests.) What about test cases that involve multiple jobs? I already see at least one spot in the code where we conditionally ignore some assertions based on test case name... you can see how those sorts of conditionals could grow over time...

I'm wondering if we can't make this more flexible and sustainable over time (and pave the way for a full battery of int tests) by structuring the test cases like this:

```
type testCase struct {
  name string
  project core.Project
  event core.Event
  assertions func(t *testing.T, eventID string, eventClient core.EventClient)
}
```

This would allow us to, case by case, say here's the project, here's the event to create for that project, and here's a function that expresses all the things I expect to be true. In each case, that function can quite easily utilize all the great "assertion" helpers you've already written here, which do a fantastic job of making complex assertions (like scanning the log stream for a specific phrase).

In the end, this might admittedly make our definition of individual test cases more verbose, but imho, it would pay dividends by making the int test "framework" more flexible and cleaner (no conditionals to accommodate specific cases).

Reply:
Thanks, definitely great foresight here.  I'll work on refactoring as illustrated with `assertions` built into the test case.  This is the predominant pattern that's been established in other tests, too, so it will be great to have alignment.

Reply:
Ok, updated.  PTAL when convenient.